ee/hogai/eval/eval_sql.py

```diff
@@ -43,9 +51,73 @@ def _run_eval_sync(self, output, expected=None, **kwargs):
         return Score(name=self._name(), score=score, metadata={"query_generation_retry_count": retry_count})
 
 
-class SQLSyntaxCorrectness(Scorer):
-    """Evaluate if the generated SQL query has correct syntax."""
+class SQLFunctionCorrectness(Scorer):
+    def _name(self):
+        return "sql_function_correctness"
+
+    def _extract_functions_from_sql(self, sql_query: str) -> set[str]:
+        """Extract all function names from SQL query using regex."""
+        if not sql_query:
+            return set()
+
+        # Pattern to match function calls: word followed by opening parenthesis
+        # This captures the function name before the opening parenthesis
+        function_pattern = r"\b([a-zA-Z_][a-zA-Z0-9_]*)\s*\("
 
+        functions = set()
+        for match in re.finditer(function_pattern, sql_query):
```

Comment:
I think this is okay, but we can just parse the SQL and extract all the `ast.Call` nodes (we do things like this in plenty of places) - this will be 100% reliable, whereas there _could_ be an edge case with this regex (though I cant quickly find one)